src/pal/src/arch/amd64/processor.cpp

```diff
@@ -54,10 +54,20 @@ Return value:
 extern "C" unsigned int XmmYmmStateSupport()
 {
     unsigned int eax;
-    __asm("  xgetbv\n" \
-        : "=a"(eax) /*output in eax*/\
-        : "c"(0) /*inputs - 0 in ecx*/\
-        : "eax", "edx" /* registers that are clobbered*/
+    __asm("  mov $1, %%eax\n" \
+          "  cpuid\n" \
+          "  and $0x18000000, %%ecx\n" /* check for xsave feature set and that it is enabled by the OS */ \
```

Comment:
A nit - if you do `xor %%eax, %%eax` right after the cpuid and at line 61 do `jne end`,  you can get rid of the jmp and xor after xgetbv.